enhancements/network/ingress-node/ingress-node-firewall-with-ebpf.md

```diff
@@ -0,0 +1,446 @@
+---
+title: ingress-node-firewall-with-ebpf
+
+authors:
+  - "@msherif1234"
+
+reviewers:
+  - "@danwinship"
+  - "@dave-tucker"
+  - "@dcbw"
+  - "@trozet"
+
+approvers:
+  - "@danwinship"
+  - "@knobunc"
+
+api-approvers:
+  - "@danwinship"
+
+creation-date: 2022-06-15
+
+last-updated: 2022-06-21
+
+status: implementable
+
+tracking-link:
+  - https://issues.redhat.com/browse/SDN-2781
+---
+
+# Ingress node Firewall using eBPF
+
+## Release Signoff Checklist
+
+- [ ] Enhancement is `implementable`
+- [ ] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Operational readiness criteria is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [openshift-docs](https://github.com/openshift/openshift-docs/)
+
+## Summary
+
+OCP customers would like to secure OCP's nodes from external attacks by configuring ingress firewall policies.
+eBPF provides a very flexible filtering mechansim to allow early detection and filtering.
```

Comment:
It might be worth calling out here that XDP in particular is where the gain is to be had, since that allows fast dropping after packet has been received but before it's been processed by the network stack.

For customers to take the best advantage of this they will want a NIC that supports XDP in HW or DRV mode.

Reply:
ok